File: 621. Task Scheduler.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        # way 1:using counter
        # freq=Counter(tasks)
        # maxfreq=freq.most_common(1)[0][1]        

        
        # way2: not using counter
        freq=collections.defaultdict(int)
        
        for item in tasks:
            freq[item]+=1
        maxfreq=0
        for val in freq.values():
            maxfreq=max (maxfreq, val)        
        
        
        
        # continue either way
        counter=0
        for k,v in freq.items():
            if v==maxfreq:  
                counter+=1
        
        res=(maxfreq-1)*(n+1)+counter        
        
        
        
        return max(len(tasks), res)

File: 621. Task Scheduler.js
var leastInterval = function(tasks, n) {
    let freq= new Map()
    for (item of tasks){
    freq.has(item)? freq.set(item,freq.get(item)+1) : freq.set(item,1)
    }
    
    let maxFreq=0
    for (val of freq.values())
       { maxFreq=Math.max(maxFreq,val)}
    
    let counter=0
    for (val of freq.values())
       {maxFreq==val && (++counter)}
 
    let res=tasks.length
    
    res=Math.max (res, (maxFreq-1)*(n+1)+counter)
    
    return res
        
    
    
    
    
    
    
};
